src/nvim/spellfile.c

```diff
@@ -2472,11 +2472,7 @@ static afffile_T *spell_read_aff(spellinfo_T *spin, char *fname)
             char buf[MAXLINELEN];
 
             aff_entry->ae_cond = getroom_save(spin, items[4]);
-            if (*items[0] == 'P') {
-              sprintf(buf, "^%s", items[4]);  // NOLINT(runtime/printf)
-            } else {
-              sprintf(buf, "%s$", items[4]);  // NOLINT(runtime/printf)
-            }
+            snprintf(buf, MAXLINELEN, *items[0] == 'P' ? "^%s" : "%s$", items[4]);
```

Comment:
sizeof(buf) is usually preferred

Reply:
I've been bitten using sizeof in the past where the argument is just a pointer.